src/Files.FullTrust/Win32API.cs

```diff
@@ -122,6 +122,35 @@ string GetDesktopAssoc()
             }
         }
 
+        public static string GetFileAssociationName(string fileName)
+        {
+            if (string.IsNullOrEmpty(fileName))
+            {
+                throw new ArgumentNullException(nameof(fileName));
+            }
+
+            uint cOut = 0;
+            var fileExt = Path.GetExtension(fileName);
+
+            if (string.IsNullOrEmpty(fileExt))
+            {
+                return null;
+            }
+
+            if (ShlwApi.AssocQueryString(ShlwApi.ASSOCF.ASSOCF_VERIFY, ShlwApi.ASSOCSTR.ASSOCSTR_FRIENDLYAPPNAME, fileExt, null, null, ref cOut) != 1)
```

Comment:
Could you check what happens if you try this on a file that has no associated program? By memory you may need to add the "ASSOCF_INIT_IGNOREUNKNOWN" flag to avoid getting wrong results. Haven't tested, does this API work when the associated app is a Store/UWP app?

Reply:
At the moment it returns "Pick an application" as a string.

Reply:
Thinking again I guess it's better to show the "Pick an application" instead of a blank space in case there's no associated program. I hadn't realized that Explorer shows that same string.